src/coreclr/vm/codeman.cpp

```diff
@@ -1467,6 +1467,36 @@ void EEJitManager::SetCpuInfo()
                                             }
                                         }
                                     }
+
+                                    if (zmmStateSupport() == 1)                                             // XGETBV XRC0[7:5] == 111
```

Comment:
This should probably be nested after the `CPUCompileFlags.Set(InstructionSet_AVX2);` for consistency with how the other checks are done and to short circuit the logic if AVX2 isn't supported.

Reply:
Changed. The `AVX512VL` logic in there is my interpretation of what you meant in terms of `AVX512*_VL` flag usage. Let me know if I misunderstood.